### DISCUSSION TITLE
Github Fork Problem

### DISCUSSION THREAD

**Discussion by code-Gambler** [Seneca-CDOT/gcc#6] [Q&A] [answered]
Hello @ctyler @MatthewDeNapoli ,

I was helping out Travis in creating a pull request to this repository, but he was not able to do so as this a fork of gcc-mirror not a standalone repository. I can create a pull request between my fork and this repo only because I have write access. I am pretty sure if we provide write access to travis, he would be able to create a pull request, but we cannot give out write accesses to all the people contributing to this repo. I found a community post which explains how to detach a fork, so that it becomes a standalone fork for experimentation purposes [here](https://github.com/orgs/community/discussions/52663). One of us (you, me or anyone with write access can do it) needs to go [here](https://support.github.com/request/fork) and do it and may be that solves the problem.

Recreation (Account used doesn't have write access to this repo)
![image](https://github.com/Seneca-CDOT/gcc/assets/98861688/60c0b1ff-8701-45d8-8506-1e1f4ff89a82)

Click on the pull request button

![image](https://github.com/Seneca-CDOT/gcc/assets/98861688/d7d77610-d257-41c5-848b-e7a46518a453)

As we can see the default base repository is gcc-mirror/gcc

![image](https://github.com/Seneca-CDOT/gcc/assets/98861688/47890afa-3cfe-4560-a26d-83c73efa9afd)

Even if we search for Seneca-CDOT's fork it doesn't show up.

Here is an example of PR from my account which has write access:

![image](https://github.com/Seneca-CDOT/gcc/assets/98861688/93160dbf-68a5-4092-8544-1e2c0ba691ee)

Please advise.

**Comment by MatthewDeNapoli** [comment]
Hello,

Make sure when creating the pull request you start off on the fork that's been created.
This is so you can specify the source as the fork Travis has created and the destination as the Seneca-CDOT repository.
If you have any other questions or need any help feel free to leave a comment.

Ex.
![image](https://github.com/Seneca-CDOT/gcc/assets/133034403/e3323606-109b-4adf-a7e3-49abab4add5a)

**Comment by code-Gambler** [comment]
I think you have write access to this fork that is the reason you are able to do it, try doing it with an account that doesn't have write access

**Comment by MatthewDeNapoli** [comment]
Phuong was able to create one and have it merged without having write access so it shouldn't be an issue with the permissions.
https://github.com/Seneca-CDOT/gcc/pull/5

**Comment by MatthewDeNapoli** [comment]
![image](https://github.com/Seneca-CDOT/gcc/assets/133034403/0d33b359-e5ca-4bbf-af5e-7a39ea0632f9)
![image](https://github.com/Seneca-CDOT/gcc/assets/133034403/d74e0458-b337-422f-bbd2-6eced10a487b)

**Comment by code-Gambler** [comment]
Phuong was able to do it, now I am confused, I really don't know what the problem is. This is my secondary account(Doesn't have write access), This is a video of me trying to create a test PR, but I am not able to locate this fork Seneca-CDOT, am I doing wrong? what do you think? same thing was happening with travis

https://github.com/Seneca-CDOT/gcc/assets/98861688/fbdcbc3e-1ff4-483b-89f6-4be235b4bb6c

**Comment by MatthewDeNapoli** [comment, marked as answer]
You need to go to the webpage of your fork and then create the pull request from there.
You're creating the pull request from the main CDOT repository which is why you're not able to select your fork.

Ex. 
Go to this link and click the "new pull request" button: https://github.com/nightmareOfDevs/gcc/pulls

**Comment by code-Gambler** [comment]
Nice, it worked! Damn never did it this for my OSD course!
Thanks, I move this issue to discussion!
